hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerSet.java

```diff
@@ -67,26 +69,28 @@ public class ContainerSet implements Iterable<Container<?>> {
       new ConcurrentSkipListMap<>();
   private final Clock clock;
   private long recoveringTimeout;
-  private final Table<ContainerID, String> containerIdsTable;
+  @Nullable
+  private final WitnessedContainerMetadataStore metadataStore;
```

Comment:
Instead of passing WitnessedContainerMetadataStore we can pass an implementation of a Table interface which would be writing to a particular table WitnessedContainerMetadataStore based on what witnessedStore updates. We can completely abstract this out so that the downstream users don't have to understand the logic.

Reply:
Raised above Jira for this handling is have real value, I do not aggree to have table being cached for long-lived object as it may cause refresh after upgrade.